### PR TITLE
[SPIR-V] Add error for rasterizer ordered view types

### PR DESCRIPTION
**Pull request by cassiebeckley**
I meant to send this a couple weeks ago. I'll be sending in a PR soon that implements this so up to you if you think it makes sense to merge this first.

### CODE REVIEW

**Review by sudonatalie — APPROVED** [review]

**Comment by sudonatalie** [comment]
> I meant to send this a couple weeks ago. I'll be sending in a PR soon that implements this so up to you if you think it makes sense to merge this first.

Either is fine with me

File: tools/clang/lib/SPIRV/AstTypeProbe.cpp [hunk 1]
```diff
@@ -1187,6 +1187,25 @@ bool isRelaxedPrecisionType(QualType type, const SpirvCodeGenOptions &opts) {
   return false;
 }
 
+bool isRasterizerOrderedView(QualType type) {
+  // Strip outer arrayness first
+  while (type->isArrayType())
+    type = type->getAsArrayTypeUnsafe()->getElementType();
```
Comment (Keenuts) [inline tools/clang/lib/SPIRV/AstTypeProbe.cpp:1193]: Curious: why use the unsafe version vs the `getAs<ArrayType>` ?

File: tools/clang/lib/SPIRV/LowerTypeVisitor.cpp [hunk 2]
```diff
@@ -630,6 +630,10 @@ const SpirvType *LowerTypeVisitor::lowerResourceType(QualType type,
 
   assert(type->isStructureOrClassType());
 
+  if (isRasterizerOrderedView(type)) {
+    emitError("rasterizer ordered views are unimplemented", srcLoc);
```
Comment (Keenuts) [inline tools/clang/lib/SPIRV/LowerTypeVisitor.cpp:634]: Running this command:
```
./build/bin/dxc -T ps_6_6 -E main -fcgl -spirv tools/clang/test/CodeGenSPIRV_Lit/type.rasterizer-ordered-buffer.unimplemented.hlsl
```

causes DXC to show the error (expected), but also the validator to complain the generated SPIR-V is invalid and to report this bug. Is that expected?

**Review by Keenuts — APPROVED** [review]
LGTM, just a question, and something probably not related to this PR specifically but a more general DXC issue.

**Comment by Keenuts** [comment]
That would be great if auto-submit wouldn't submit if there was unresolved comments 🙃

**Pull request merged**